README.md

```diff
@@ -27,6 +27,49 @@ This extension can be configured using following parameters:
     * 1d - one day 
     * 2d 5h 30m - two days, five hours and thirty minutes 
 
+## JWT Payload translation
+The bigcommerce.user.autologin.v1 pipeline was created to log the customer in our App from a BigCommerce in-app browser. 
+It is called by a theme script which is injected into the BigCommerce store via Web Analytics Google script.
+The theme generator code can be found [here](https://github.com/shopgate/bigcommerce-js) as well as the instructions.
+
+One can also notice that the customer is requested from the theme 
+[code](https://github.com/shopgate/bigcommerce-js/blob/ff923770a1842184547413425b909b27a103c071/src/modules/app_event_subscribers/autologin.js#L20) 
+and passed into our autologin pipeline.
+
+To increase security we outsourced the translation of this JWT payload token to a different plugin, 
+see the mentioned pipeline file to see how it is used:
+```$xslt
+{
+    "type": "pipeline",
+    "id": "bigcommerce.user.customerJwtDecrypt.v1.json",
+    "input": [
+      {"key": "payload", "id": "200"},
+      {"key": "algorithm", "id": "300", "optional": true}
+    ],
+    "output": [
+      {"key": "customer", "id": "1000"}
+    ]
+},
+{
+    "type": "errorCatchExtension",
+    "id": "@shopgate/bigcommerce-user",
+    "path": "@shopgate/bigcommerce-user/lib/errors/catchJwtError.js",
+    "input": [],
+    "output": []
+},
+```
+Note that all customerJwtDecrypt does is decrypting the payload via an 
+npm module [jwt-simple](https://www.npmjs.com/package/jwt-simple):
```

Comment:
Should this be "a npm module" instead of "an npm module"?

Reply:
can be, for me an sounds more correct, but looking into grammar -> a is "more" correct

Reply:
according to grammar, "an" is used if the following word starts with a vowel sound. So to me "an npm module" looks perfectly fine to me

Reply:
well, based on what you said `npm` is a consonant sounding word (it has no vowels) then `a npm` is more correct